src/proxy/processors/push-action/metrics.ts

```diff
@@ -0,0 +1,37 @@
+import { Step } from '../../actions';
+
+/**
+ * Performance Timer
+ *
+ * Logs basic timing info for operations
+ */
+export class PerformanceTimer {
+  private step: Step;
+  private startTime: number = 0;
+  private operation: string = '';
+
+  constructor(step: Step) {
+    this.step = step;
+  }
+
+  start(operation: string): void {
+    this.operation = operation;
+    this.startTime = Date.now();
```

Comment:
Can this class be adjusted to use `performance.now()` for safer reporting

https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
https://nodejs.org/api/perf_hooks.html#performancenow
https://stackoverflow.com/questions/111368/how-do-you-performance-test-javascript-code